### PR TITLE
grant/cs/prepopulate datatype map

### PR DESCRIPTION
**Pull request by grant-Kolena**
Based on @consoull's work [here](https://github.com/kolenaIO/kolena/pull/570) 

### Linked issue(s)
Fixes KOL-5860

### What change does this PR introduce and why?
Fixes a bug where `_DATA_TYPE_MAP` is missing Annotation, Asset, TestSample, Thresholded if the corresponding `TypedDataObject` has not been subclassed in the current Python process.

### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: kolena/_utils/datatypes.py [hunk 1]
```diff
@@ -77,13 +79,44 @@ def _allow_extra(cls: Type[T]) -> bool:
...
+
 def _get_full_type(obj: Type["TypedDataObject"]) -> str:
     data_type = obj._data_type()
-    return f"{data_type._data_category()}/{data_type.value}"
+    return f"{data_type._data_category().value}/{data_type.value}"
 
 
 def _get_data_type(name: str) -> Optional[Type["TypedDataObject"]]:
-    return _DATA_TYPE_MAP.get(name, None)
+    class_type = _DATA_TYPE_MAP.get(name, None)
```
Comment (ly-kolena) [inline kolena/_utils/datatypes.py:110]: Can we add a test to make sure the original bug will be fixed?

Comment (grant-Kolena) [reply kolena/_utils/datatypes.py:110]: yes will add a test

File: kolena/_utils/datatypes.py [hunk 2]
```diff
@@ -188,7 +222,7 @@ def deserialize_field(
             field_type = field_type or field.type
             origin = get_origin(field_type)  # non-None for typing.X types
 
-            if origin is list:
+            if origin is list or origin is Sequence:
```
Comment (grant-Kolena) [inline kolena/_utils/datatypes.py:225]: deserialization was failing for `NumberSeries = Sequence[Union[float, int]]` without it

File: kolena/_utils/datatypes.py [hunk 3]
```diff
@@ -262,23 +296,9 @@ def _serialize_dataobject(x: Any) -> Any:
     return x._to_dict() if isinstance(x, DataObject) else x
 
 
-def _deserialize_dataobject(x: Any) -> Any:
```
Comment (grant-Kolena) [inline kolena/_utils/datatypes.py:265]: this is a duplicated method

File: tests/unit/workflow/test_plot.py [hunk 4]
```diff
@@ -97,7 +97,7 @@ def test__curve_plot__serialize() -> None:
         "x_label": "x",
         "y_label": "y",
         "curves": [{"label": "a", "x": [1, 2, 3], "y": [2, 3, 4], "extra": None}],
-        "data_type": f"{_PlotType._data_category()}/{_PlotType.CURVE.value}",
+        "data_type": f"{_PlotType._data_category().value}/{_PlotType.CURVE.value}",
```
Comment (ly-kolena) [inline tests/unit/workflow/test_plot.py:100]: are these changes needed?

Comment (grant-Kolena) [reply tests/unit/workflow/test_plot.py:100]: the result from the string formatter should be the same as we do extend `str` class as well, but it breaks the CI without `.value`

File: kolena/_utils/datatypes.py [hunk 5]
```diff
@@ -296,5 +316,11 @@ def _data_type() -> U:
...
         self_dict = super()._to_dict()
         self_type = self._data_type()
-        self_dict[DATA_TYPE_FIELD] = f"{self_type._data_category()}/{self_type.value}"
+        self_dict[DATA_TYPE_FIELD] = f"{self_type._data_category().value}/{self_type.value}"
         return self_dict
+
+    def __init_subclass__(cls, **kwargs: Any):
+        try:
+            _register_data_type(cls)
+        except NotImplementedError:
```
Comment (ly-kolena) [inline kolena/_utils/datatypes.py:325]: nit: when would this exception be triggered?

Comment (grant-Kolena) [reply kolena/_utils/datatypes.py:325]: Things like the the Annotation [class](https://github.com/kolenaIO/kolena/blob/39f715988e45997a30ad361da648b46aef056a90/kolena/annotation.py#L65), this is the parent of BoundingBox, etc that extends TypedDataObject as well

**Review by ly-kolena — APPROVED** [review]
